src/node_contextify.cc

```diff
@@ -288,6 +288,15 @@ BaseObjectPtr<ContextifyContext> ContextifyContext::New(
             .IsNothing()) {
       return BaseObjectPtr<ContextifyContext>();
     }
+
+    if (new_context_global
```

Comment:
I think we only need to set this on the wrapper? And in the import callback you can get to the wrapper via `ContextEmbedderIndex::kContextifyContext`

Reply:
Yes, it works. However, if the field is defined on the global object, the import callback can generalize the condition for principal realms, shadow realms, and contextified contexts by getting the referrer symbol from `context->Global()`. So I believe the current form might be more straightforward in the dynamic import callback.

Reply:
Good point, can you add a comment here explaining why this we set this in the global as well?

I imagine for realms you can also do `ContextEmbedderIndex::kRealm` and make this a per-realm property. Though using a private symbol could open the doors to embedder contexts that have neither slots.

Reply:
Thanks for the suggestion, added a comment to explain that. Using a per-realm property is also a good path. As mentioned, it looks more generic when using a private symbol on the global object.